quantumdb-cli/src/main/java/io/quantumdb/cli/xml/XmlMapper.java

```diff
@@ -75,7 +75,20 @@ public void characters(char[] ch, int start, int length) {
 
 					if (stack.size() > 0) {
 						XmlElement parent = stack.get(stack.size() - 1);
-						parent.getChildren().add(element);
+						if (parent.getTag() == null) {
+							element = new XmlElement(null, parent.getText().concat(body));
+							element.getChildren().addAll(parent.getChildren());
+							element.getAttributes().putAll(parent.getAttributes());
+							if (stack.size() > 1) {
+								XmlElement grandParent = stack.get(stack.size() - 2);
+								grandParent.getChildren().remove(parent);
+								grandParent.getChildren().add(element);
+							}
+							stack.remove(parent);
+						}
+						else {
+							parent.getChildren().add(element);
+						}
```

Comment:
Took me a whole afternoon to arrive at this solution. Other solutions where a StringBuilder was used to append strings and make 1 new XmlElement in the endElement function caused errors.